### ISSUE TITLE
error in typescript.js - os.platform() is not a function

### ISSUE THREAD

**Issue by frankwallis** [jspm/jspm-cli#1092]
When I do ```jspm install typescript@1.6.0-beta```, typescript.js contains the following:

```
      function getNodeSystem() {
        var _fs = require('@empty');
        var _path = require('@empty');
        var _os = require('@empty');
        var platform = _os.platform();
```

And when I run jspm bundle it throws the error

```
Error: TypeError: file:///Users/frank/work/plugin-typescript/example/angular2/jspm_packages/npm/typescript@1.6.0-beta/lib/typescript.js:1533
        var platform = _os.platform();
                           ^
_os.platform is not a function
	Evaluating file:///Users/frank/work/plugin-typescript/example/angular2/jspm_packages/npm/typescript@1.6.0-beta/lib/typescript.js
```

Any ideas?

**Comment by guybedford** [comment]
Thanks for letting me know, I was able to resolve this via `jspm install typescript@1.6.0-beta -o "{map:{fs:'@empty',path:'@empty'}}"`. I've added the override in https://github.com/jspm/registry/blob/master/package-overrides/npm/typescript@1.6.0-beta.json which should have the same affect.

**Comment by guybedford** [comment]
@frankwallis it seems that typescript works fine in jspm with the override `jspm install typescript@1.6.0-beta -o "{map:{fs:'@empty',path:'@empty'}}"` and is somewhat smaller as well. If you feel like testing if that works fine even when building, could be useful to get the footprint down.

**Comment by frankwallis** [comment]
Thanks, yes it works with the ```browser: {}``` override and when I install with ```-o "{map:{fs:'@empty',path:'@empty'}}"```. When using the manual override it looks like it is also applying the browser one from the registry so it is unclear if the browser one is still needed or not. I haven't used these settings before, what do they actually do?

If you are making changes then it is probably worth adding typescript@1.7.0-dev.json to the registry as that is the name of the current overnight build. I can send a pull request if you prefer.

Many thanks for your help.

**Comment by guybedford** [comment]
@frankwallis the map values for `fs` and `path` will stop jspm from unnecessarily downloading `fs` and `path` browserify libs, even if it thinks the npm package would need them. So it should cut down on the install size. The problem is if TypeScript actually tries to use `fs` or `path` it will get an empty module, hence my asking if we can get away with it or not. Effectively this is a way to reduce the config file size for npm packages, which is a common question for jspm users installing things from npm.

A PR for the next TypeScript version would be amazing. I've created https://github.com/jspm/jspm-cli/issues/1101 to avoid this issue in future.

**Comment by frankwallis** [comment]
I have created https://github.com/jspm/registry/pull/566

It looks to me like TypeScript does not require any dependencies once 'browser' has been defined, so I have added overrides for all of its dependencies. I have tested with the plugin and the built in compiler and it browser loads and bundles fine.

I imagine this is how it runs in the [TypeScript Playground](http://www.typescriptlang.org/Playground), when it is browser hosted.
